src/odemis/driver/test/xt_client_test.py

```diff
@@ -281,6 +283,171 @@ def test_set_beam_shift(self):
+            time.sleep(2.5)
+
+            # Stop auto auto contrast brightness and check if it stopped running.
+            auto_contrast_brightness_future.cancel()
+            time.sleep(5.0)  # Give microscope/simulator the time to update the state
+            auto_contrast_brightness_state = self.microscope.is_running_auto_contrast_brightness(channel)
+            self.assertEqual(auto_contrast_brightness_state, False)
+            self.assertIsInstance(auto_contrast_brightness_future, ProgressiveFuture)
+
+            # Test starting auto contrast brightness and cancelling directly
+            auto_contrast_brightness_future = self.scanner.applyAutoContrastBrightness(role)
+            auto_contrast_brightness_future.cancel()
+            time.sleep(5.0)  # Give microscope/simulator the time to update the state
+            auto_contrast_brightness_state = self.microscope.is_running_auto_contrast_brightness(channel)
+            self.assertEqual(auto_contrast_brightness_state, False)
+            self.assertIsInstance(auto_contrast_brightness_future, ProgressiveFuture)
+
+            # Start auto contrast brightness
+            max_execution_time = 60  # Approximately 3 times the normal expected execution time (s)
+            starting_time = time.time()
+            auto_contrast_brightness_future = self.scanner.applyAutoContrastBrightness(role)
+            time.sleep(0.5)  # Give microscope/simulator the time to update the state
+            # Wait until the auto contrast brightness is finished
+            auto_contrast_brightness_future.result(timeout=max_execution_time)
+
+            # Check if the time to perform the auto contrast brightness is not too long
+            self.assertLess(time.time() - starting_time, max_execution_time,
+                            "Execution of auto contrast brightness was stopped because it took more than %s seconds."
+                            % max_execution_time)
+
+            auto_contrast_brightness_state = self.microscope.is_running_auto_contrast_brightness(channel)
+            self.assertEqual(auto_contrast_brightness_state, False)
+
+        # Test if an error is raised when an invalid detector is provided
+        with self.assertRaises(KeyError):
+            self.scanner.applyAutoContrastBrightness("error_expected")
+        time.sleep(2.5)  # Give microscope/simulator the time to update the state
+        autofocus_state = self.microscope.is_autofocusing(channel)
+        self.assertEqual(autofocus_state, False)
+
+    def test_apply_autofocus(self):
+        """
+        Test for the auto functionality of the autofocus.
+        """
+        # Check all the different detector types by looping over them
+        for role, channel in DETECTOR2CHANNELNAME.items():
+            # Start auto focus and check if it is running.
+            autofocus_future = self.efocus.applyAutofocus(role)
+            autofocus_state = self.microscope.is_autofocusing(channel)
```

Comment:
Wait a tiny bit before checking.

Reply:
If already in focus or with a correct brighness/contrast these methods can run really fast. For auto contrast brigthness this can be < 1 sec for autofocus this can be up to around 1 sec. We can implement a wait but we then have the chance of a false fail of the test because of the wait. Currently the status update is fast enough and the tests will pas.
What do you think I should do?

Reply:
That works indeed.